transform-plugins/src/main/java/co/cask/hydrator/plugin/ValueMapper.java

```diff
@@ -0,0 +1,228 @@
+        if (fieldsMapping.containsKey(inputField.getName())) {
+          outputFields.add(Schema.Field.of(fieldsMapping.get(inputField.getName()),
+                  Schema.of(Schema.Type.STRING)));
+        } else {
+          outputFields.add(inputField);
+        }
+      }
+    }
+    outputSchema = Schema.recordOf(inputSchema.getRecordName() + ".formatted", outputFields);
+    schemaCache.put(inputSchema, outputSchema);
+    return outputSchema;
+  }
+
+  /**
+   * Parse configuration provided by user
+   */
+  public void parseConfiguration(String mapping, TransformContext context) {
+    LOG.debug("in parseConfiguration:" + mapping);
+    JsonParser jsonParser = new JsonParser();
+    JsonElement jsonMapping = jsonParser.parse(mapping);
+    LOG.debug("in parseConfiguration:" + jsonMapping.toString());
+    mappingArray = jsonMapping.getAsJsonObject().get("mapping").getAsJsonArray();
+    for (JsonElement mappingElement : mappingArray) {
+      String sourceField = mappingElement.getAsJsonObject().get("mapperField").getAsString();
+      String targetField = mappingElement.getAsJsonObject().get("mappingField").getAsString();
+      fieldsMapping.put(sourceField, targetField);
+      ValueMapperLookUp lookUpTable = init(context, mappingElement.getAsJsonObject().get("lookup").toString());
+      datasetMapping.put(sourceField, lookUpTable);
+    }
+  }
+
+  /**
+   * Creates a look up table
+   */
+  private ValueMapperLookUp init(LookupProvider context, String lookUp) throws IllegalArgumentException {
+
+    LOG.debug("in init:" + lookUp);
+    LookupConfig lookUpConfig;
+    try {
+      lookUpConfig = GSON.fromJson(lookUp, LookupConfig.class);
+    } catch (JsonSyntaxException e) {
+      throw new IllegalArgumentException("Invalid lookup config. Expected map of string to string", e);
+    }
+
+    ValueMapperLookupProvider valueMapperLookupProvider = new ValueMapperLookupProvider(context, lookUpConfig);
+    String lookUpTable = parseLookUpTableName(lookUp);
+
+    ValueMapperLookUp lookupTable = valueMapperLookupProvider.provide(lookUpTable);
+    return lookupTable;
+  }
+
+  /**
+   * parse lookup table name from lookup object
+   */
+  private String parseLookUpTableName(String lookUp) {
+    String tableName = null;
+    JsonParser jsonParser = new JsonParser();
+    JsonElement jsonMapping = jsonParser.parse(lookUp);
+    JsonElement table = jsonMapping.getAsJsonObject().get("tables");
+    for (Map.Entry entry : table.getAsJsonObject().entrySet()) {
+      tableName = entry.getKey().toString();
+      break;
+    }
+    return tableName;
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    parseConfiguration(config.mapping, context);
+  }
+
+  /**
+   * Configuration for the ValueMapper transform.
+   */
+  public static class Config extends PluginConfig {
+
+    //TODO input parameters may be changed after having three different input params from UI
```

Comment:
We have already mentioned it as a risk that we don’t understand the inputs and outputs structure for the plugins. 
However, discussed it with Sree, and he says widget construction is already in progress. Suggestion was to use a temporary widget and proceed with the same